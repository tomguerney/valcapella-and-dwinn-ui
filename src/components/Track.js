import React from 'react'
import Card from '@material-ui/core/Card'
import CardActionArea from '@material-ui/core/CardActionArea'
import Collapse from '@material-ui/core/Collapse'
import Typography from '@material-ui/core/Typography'
import AudioPlayer from './AudioPlayer'

export default class Track extends React.Component {
  render() {
    return (
      <Card
        style={{
          margin: 20,
          backgroundImage: 'url(' + this.props.backgroundImage + ')',
          backgroundSize: 'cover',
        }}>
        <CardActionArea disableRipple onClick={() => this.props.onTrackSelected(this.props.id)}>
          <Typography
            style={{ padding: 30, fontFamily: 'cinzel decorative' }}
            component='h1'
            variant='h2'
            align='center'>
            {this.props.title}
          </Typography>
        </CardActionArea>
        <Collapse in={this.props.openTrackId === this.props.id}>
          <AudioPlayer
            audioSource={this.props.audioSource}
            audioHidden={this.props.id !== this.props.openTrackId}
          />
        </Collapse>
      </Card>
    )
  }
}
