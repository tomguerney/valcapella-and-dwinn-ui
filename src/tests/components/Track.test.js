import React from 'react'
import { shallow } from 'enzyme'
import Track from '../../components/Track'

let wrapper, onTrackSelected, id

beforeEach(() => {
  id = 'track1'
  onTrackSelected = jest.fn()
  wrapper = shallow(
    <Track
      id={id}
      title='Act 1'
      backgroundImage='background_image.png'
      audioSource='audio_source.mp3'
      onTrackSelected={onTrackSelected}
      openTrackId='notrack'
    />
  )
})

test('should render Track collapsed correctly', () => {
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('AudioPlayer').prop('audioHidden')).toBe(true)
  expect(wrapper.find('WithStyles(ForwardRef(Collapse))').prop('in')).toBe(false)
})

test('should render Track expanded correctly', () => {
  wrapper.setProps({ openTrackId: id })
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('AudioPlayer').prop('audioHidden')).toBe(false)
  expect(wrapper.find('WithStyles(ForwardRef(Collapse))').prop('in')).toBe(true)
})

test('should handle onTrackSelected correctly', () => {
  wrapper.find('WithStyles(ForwardRef(CardActionArea))').simulate('click')
  expect(onTrackSelected).toHaveBeenCalledWith(id)
})
