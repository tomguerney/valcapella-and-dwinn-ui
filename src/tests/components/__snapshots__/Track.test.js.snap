// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Track collapsed correctly 1`] = `
<WithStyles(ForwardRef(Card))
  style={
    Object {
      "backgroundImage": "url(background_image.png)",
      "backgroundSize": "cover",
      "margin": 20,
    }
  }
>
  <WithStyles(ForwardRef(CardActionArea))
    disableRipple={true}
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(Typography))
      align="center"
      component="h1"
      style={
        Object {
          "fontFamily": "cinzel decorative",
          "padding": 30,
        }
      }
      variant="h2"
    >
      Act 1
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(Collapse))
    in={false}
  >
    <AudioPlayer
      audioHidden={true}
      audioSource="audio_source.mp3"
    />
  </WithStyles(ForwardRef(Collapse))>
</WithStyles(ForwardRef(Card))>
`;

exports[`should render Track expanded correctly 1`] = `
<WithStyles(ForwardRef(Card))
  style={
    Object {
      "backgroundImage": "url(background_image.png)",
      "backgroundSize": "cover",
      "margin": 20,
    }
  }
>
  <WithStyles(ForwardRef(CardActionArea))
    disableRipple={true}
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(Typography))
      align="center"
      component="h1"
      style={
        Object {
          "fontFamily": "cinzel decorative",
          "padding": 30,
        }
      }
      variant="h2"
    >
      Act 1
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(Collapse))
    in={true}
  >
    <AudioPlayer
      audioHidden={false}
      audioSource="audio_source.mp3"
    />
  </WithStyles(ForwardRef(Collapse))>
</WithStyles(ForwardRef(Card))>
`;
